clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2507,32 +2521,35 @@ public void testCurrentLag(GroupProtocol groupProtocol) {
         assertEquals(OptionalLong.of(45L), consumer.currentLag(tp0));
     }
 
-    // TODO: this test triggers a bug with the CONSUMER group protocol implementation.
-    //       The bug will be investigated and fixed so this test can use both group protocols.
     @ParameterizedTest
-    @EnumSource(value = GroupProtocol.class, names = "CLASSIC")
+    @EnumSource(GroupProtocol.class)
     public void testListOffsetShouldUpdateSubscriptions(GroupProtocol groupProtocol) {
         final ConsumerMetadata metadata = createMetadata(subscription);
         final MockClient client = new MockClient(time, metadata);
 
         initMetadata(client, singletonMap(topic, 1));
 
-        consumer = newConsumer(groupProtocol, time, client, subscription, metadata, assignor, true, groupInstanceId);
-
+        consumer = newConsumer(groupProtocol, time, client, subscription, metadata, assignor, false,
+                null, groupInstanceId, false);
         consumer.assign(singleton(tp0));
-
-        // poll once to update with the current metadata
-        consumer.poll(Duration.ofMillis(0));
-        client.respond(FindCoordinatorResponse.prepareResponse(Errors.NONE, groupId, metadata.fetch().nodes().get(0)));
-
         consumer.seek(tp0, 50L);
-        client.prepareResponse(listOffsetsResponse(singletonMap(tp0, 90L)));
 
+        client.prepareResponse(listOffsetsResponse(singletonMap(tp0, 90L)));
```

Comment:
Can `client.prepareResponse` be moved to `newConsumer`?

Reply:
@FrankYang0529 could you address this? I guess the suggestion comes from the recent fix to some tests that were flaky for a similar situation (https://github.com/apache/kafka/pull/170560)

Reply:
Hi @lianetm and @TaiJuWu, thanks for the suggestion, but I think we can keep this line just before `consumer.endOffsets`.

In `AsyncKafkaConsumer#endOffsets`, it sends `ListOffsetsEvent` and `ApplicationEventProcessor#process(ListOffsetsEvent)` calls `OffsetsRequestManager#fetchOffsets`.

In `OffsetsRequestManager#fetchOffsets`, it builds the request and `AsyncKafkaConsumer#endOffsets` uses `ApplicationEventHandler#addAndGet` to wait for the result, so I think it's safe to put it just before `consumer.endOffsets`. WDYT? Thank you.

Reply:
Thanks for your explanations.

Reply:
makes sense, but I still have a concern with this (wonder if it would make this test flaky). Let's say the moment we get to prepare this response, there is another request that got generated (a fetch request, for instance). Wouldn't that make that the endOffsets request will never find the response? (Not sure if I'm missing details of how prepareResponse works)

Reply:
for the record, I was concerned here mostly about fetch requests (since we have a partition assigned and not paused), but just noticed we're not calling consumer.poll (so we shouldn't expect FETCH requests to be generated)

Still, worth mentioning that without this in-flight fix https://github.com/apache/kafka/pull/17035 for Fetch, we could wrongly generate fetch requests even without polling I guess. So wonder if enabling this test like this would be flaky for the new consumer without that fix.